### DISCUSSION TITLE
How best to redirect all requests to '/' to '/somethingelse/'?

### DISCUSSION THREAD

**Discussion by rorm** [vercel/next.js#23235] [Help]
Apologies for what's probably an inane question—I'm new to all this—but I was wondering how I should go about making sure that everyone who visits the root of my site will be immediately redirected to a subpath? I want my root to be `mysite.com/somethingelse/`.

 I bought a domain like `firstnames.ir/name/` and now regret it.

**Comment by litcodeur** [comment]
The best way of redirecting in your case would be via SSR on the index page so that there is no flash of any content on the user screen by including the following code in the index page.
```
export const getServerSideProps: GetServerSideProps = async () => {
	return { redirect: { destination: '/name', permanent: true } };
};
```


However I highly doubt that you could register a domain name like you mentioned as most providers wouldn't allow such things. You might want to double check with the provider and you might not need to include this after all and all users on the website could shown the root at index.
